models/base/manifest/default/snowplow_mobile_base_sessions_lifecycle_manifest.sql

```diff
@@ -0,0 +1,89 @@
+{{ 
+  config(
+    materialized=var("snowplow__incremental_materialization"),
+    unique_key='session_id',
+    upsert_date_key='start_tstamp',
+    partition_by = {
+      "field": "start_tstamp",
+      "data_type": "timestamp"
+    },
+    cluster_by=mobile_cluster_by_fields_sessions_lifecycle(),
+    full_refresh=snowplow_mobile.allow_refresh(),
+    tags=["manifest"]
+  ) 
+}}
+
+{% set lower_limit, upper_limit, _ = snowplow_utils.return_base_new_event_limits(ref('snowplow_mobile_base_new_event_limits')) %}
+{% set session_lookback_limit = snowplow_utils.get_session_lookback_limit(lower_limit) %}
+{% set is_run_with_new_events = snowplow_utils.is_run_with_new_events('snowplow_mobile') %}
+{% set session_id = snowplow_mobile.get_session_id_path_sql(relation_alias='e') %}
+{% set user_id  = snowplow_mobile.get_device_user_id_path_sql(relation_alias='e')%}
+
+with new_events_session_ids as (
+  select
+    {{ session_id }} as session_id,
+    max( {{ user_id }} ) as device_user_id,
+    min(e.collector_tstamp) as start_tstamp,
+    max(e.collector_tstamp) as end_tstamp
+
+  from {{ source('atomic', 'events') }} e
+
+  where
+    {{ session_id }} is not null
+    and e.dvce_sent_tstamp <= {{ snowplow_utils.timestamp_add('day', var("snowplow__days_late_allowed", 3), 'dvce_created_tstamp') }} -- don't process data that's too late
+    and e.collector_tstamp >= {{ lower_limit }}
+    and e.collector_tstamp <= {{ upper_limit }}
+    and {{ snowplow_utils.app_id_filter(var("snowplow__app_id",[])) }}
+    and e.platform in ('{{ var("snowplow__platform")|join("','") }}') -- filters for 'mob' by default
+    and {{ is_run_with_new_events }} --don't reprocess sessions that have already been processed.
+    {% if var('snowplow__derived_tstamp_partitioned', true) and target.type == 'bigquery' | as_bool() %} -- BQ only
+      and cast(e.derived_tstamp as datetime) >= {{ dbt_utils.dateadd('hour', -1 * var('snowplow__lookback_window_hours', 6), lower_limit) }}
```

Comment:
Thanks for the tip Will, I've updated the lines to be using `timestamp_add` from the `snowplow_utils` package. I've read the link and I agree it states that we shouldn't be dynamically filtering on dates, however on running the two (one with dynamically calculated dates and the other with fixed dates), there's no noticeable difference in any of the runtimes or memory usage.
Here you can see the results with dates hardcoded (between `2021-03-21` and `2021-04-21`)
![image](https://user-images.githubusercontent.com/2262032/154771422-eaabee80-3f67-4518-85f3-ca3664e3faeb.png)

versus the same but using the `timestamp_add` function:
![image](https://user-images.githubusercontent.com/2262032/154771538-4fcf04b9-9709-4649-880a-2c0a3d795695.png)

I reckon it might have to do with the `collector_tstamp` filter which is based on a harcoded timestamp, and therefore I would imagine the partitioning occurs there.

Reply:
Nice thanks for updating. Just to demonstrate the difference: 
Casting then filtering:
![image](https://user-images.githubusercontent.com/71764352/154986474-7a64e3d8-bf8e-4862-b01a-a022e9018067.png)
The double filter on `collector_tstamp` is simply because the events table requires a filter on this columns, and BQ doesn't recognise the casted column as such a filter.
No casting:
![image](https://user-images.githubusercontent.com/71764352/154986631-a3995803-7553-45c6-bf13-74b7c79d4bff.png)

Note the difference in GB scanned.

You are right though, the dynamic timestamp calc doesnt impact partition pruning:

![image](https://user-images.githubusercontent.com/71764352/154987349-335ec858-fec9-40b5-b0a6-2117b727dd4c.png)
vs.
![image](https://user-images.githubusercontent.com/71764352/154987409-80b633d9-5a4b-465e-b9d9-0c8caae38a91.png)